R/otel.R

```diff
@@ -420,8 +420,19 @@ local_promise_domain <- function(
   }
 
   # Inspired by `withr::defer()`
+  defer(
+    {
+      globals$domain <- oldval
+    },
+    envir = envir
+  )
+
+  invisible()
+}
+
+defer <- function(expr, envir = parent.frame()) {
```

Comment:
For easier maintenance, suggest we replace with `defer()` from https://github.com/r-lib/withr/blob/main/R/standalone-defer.R verbatim.
Or perhaps: `usethis::use_standalone("r-lib/withr", file = "defer")`.

Notice also that you set `after = TRUE`, whereas this is default `FALSE` in these situations.